src/set_theory/ordinal/natural_ops.lean

```diff
@@ -0,0 +1,354 @@
+/-
+Copyright (c) 2022 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+
+import set_theory.ordinal.arithmetic
+
+/-!
+# Natural operations on ordinals
+
+The goal of this file is to define natural addition and multiplication on ordinals, and provide a
+basic API. The natural addition of two ordinals `a + b` is recursively defined as the least ordinal
+greater than `a' + b` and `a + b'` for `a' < a` and `b' < b`. The natural multiplication `a * b` is
+likewise recursively defined as the least ordinal such that `a * b + a' * b'` is greater than
+`a' * b + a * b'` for any `a' < a` and `b' < b`.
+
+These operations form a rich algebraic structure: they're commutative, associative, have the usual
+`0` and `1` from ordinals, and distribute over one another.
+
+Moreover, these operations are the addition and multiplication of ordinals when viewed as `surreal`
+numbers. This makes them particularly useful for game theory.
+
+Finally, both operations admit simple, intuitive descriptions in terms of the Cantor normal form.
+The natural addition of two ordinals corresponds to adding the Cantor normal forms as if they were
+polynomials in `ω`. Likewise, their natural multiplication corresponds to multiplying the Cantor
+normal forms as polynomials.
+
+# Implementation notes
+
+Given the rich algebraic structure of these two operations, we choose to implement them via a type
+synonym `nat_ordinal`. We provide many auxiliary theorems to make the transition between `ordinal`
+and `nat_ordinal` as seamless as possible.
+
+# Todo
+
+- Define natural multiplication and provide a basic API.
+- Prove the characterizations of natural addition and multiplication in terms of the Cantor normal
+  form.
+-/
+
+universes u v
+
+open function ordinal
+
+noncomputable theory
+
+/-- A type synonym for ordinals with natural addition and multiplication. -/
+def nat_ordinal : Type* := ordinal
+
+namespace nat_ordinal
+
+/-- The identity function between `ordinal` and `nat_ordinal`. -/
+@[pattern] def to_nat_ordinal : ordinal ≃ nat_ordinal := equiv.refl _
+
+/-- The identity function between `nat_ordinal` and `ordinal`. -/
+@[pattern] def of_nat_ordinal : nat_ordinal ≃ ordinal := equiv.refl _
```

Comment:
These names are a bit verbose; this is called `nat_ordinal.to_nat_ordinal` and `nat_ordinal.of_nat_ordinal`.

One reasonable idea would be to call these `ordinal.to_nat_ordinal` and `nat_ordinal.to_ordinal`, which would also have the bonus of making dot notation work.

We don't have that luxury with most type aliases as the the alias is usually not for a concrete type. Here we do, so I think diverging from the pattern set by `multiplicative.of_add` and `order_dual.to_dual` is fine.

Reply:
I've renamed them accordingly.